### PR TITLE
Initial commit of testtool.

### PR DESCRIPTION
**Pull request by petermattis**
Eases creation and management of docker clusters and test binaries.

### CODE REVIEW

**Comment by petermattis** [comment]
This is something that started as shell scripts to talk to google compute engine. I've recently converted it to go and switched it to using docker-machine. I don't expect a detailed review, but worth taking a look at what I'm using to start VMs on google compute engine and build binaries for running there.

**Comment by MBerhault** [comment]
ah. fun. I'm doing pretty much the same thing for aws, although it will be pretty easy to add new drivers (same meaning of driver that docker-machine has).
https://github.com/cockroachdb/cockroach-prod/tree/marc/aws_stuff

File: add.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/codegangsta/cli"
```
Comment (bdarnell) [inline add.go]: Why did you use this library instead of go-commander (which we are using in the main binary)?

Comment (petermattis) [inline add.go]: It has subcommands and some other stuff that seemed interesting, though I ended up not using subcommands. The go-commander library looks abandoned (still on code.google.com).

**Comment by tschottdorf** [comment]
LGTM. Would be fun if the whole team had some GCE at their disposal to play around with this as it grows.

**Pull request merged**